find-channel/src/test/java/no/unit/nva/channel/ChannelRegistryClientTest.java

```diff
@@ -73,7 +73,24 @@ public void testValidResponse() throws IOException, NoResultsFoundException, URI
                 "http://example.org");
         List<Channel> channels = channelRegistryClient.fetchChannels(851, "searchTerm");
 
-        Assert.assertEquals(10, channels.size());
+        Assert.assertEquals(8, channels.size());
+    }
+
+    @Test
+    public void channelRegistryResponseReturnsOneChannelWhenInputMatchesOverlappingJournals() throws IOException,
+            URISyntaxException, NoResultsFoundException {
+        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
+        CloseableHttpResponse response = mock(CloseableHttpResponse.class);
+
+        when(httpClient.execute(any())).thenReturn(response);
+        when(response.getEntity()).thenReturn(new StringEntity(new String(Files.readAllBytes(Paths.get(getClass()
```

Comment:
Maybe extract some code to methods

Reply:
You can also assign the expected and the actual values in variables.